storage_drivers/azure/api/azure.go

```diff
@@ -129,17 +131,49 @@ type Client struct {
 }
 
 // NewDriver is a factory method for creating a new SDK interface.
-func NewDriver(config ClientConfig) (Azure, error) {
+func NewDriver(ctx context.Context, config ClientConfig) (Azure, error) {
 	var err error
+	var credential azcore.TokenCredential
 
 	// Ensure we got a location
 	if config.Location == "" {
 		return nil, errors.New("location must be specified in the config")
 	}
 
-	credential, err := azidentity.NewClientSecretCredential(config.TenantID, config.ClientID, config.ClientSecret, nil)
-	if err != nil {
-		return nil, err
+	if credFilePath := os.Getenv("AZURE_CREDENTIAL_FILE"); credFilePath != "" {
```

Comment:
what if `AZURE_CREDENTIAL_FILE` not defined? default value should be `/etc/kubernetes/azure.json`

Reply:
The `AZURE_CREDENTIAL_FILE` env is injected by trident operator, it won't be empty if it is deployed on AKS, otherwise it is not running on AKS.